src/npc.cpp

```diff
@@ -1240,8 +1240,7 @@ void npc::say( const std::string &line, const int priority ) const
         add_msg( m_warning, _( "%1$s says something but you can't hear it!" ), name );
     }
     // Hallucinations don't make noise when they speak
-    if( !is_hallucination() ) {
```

Comment:
Wasn't the solution to change the check to `if (is_hallucination)` and keep the `add_msg` so that hallucination speak without noise and real NPC make noise?